### PR TITLE
[Fix] Fix load_ckpt from pavi and s3

### PR DESCRIPTION
**Pull request by LXXXXR**
Hi,
This pull request fixes load checkpoint from pavi and s3. All ranks download ckpt now. Pls kindly take a look.Thank you.

### CODE REVIEW

File: mmcv/runner/checkpoint.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def load_from_http(filename, map_location=None, model_dir=None):
 @CheckpointLoader.register_scheme(prefixes='pavi://')
 def load_from_pavi(filename, map_location=None):
     """load checkpoint through the file path prefixed with pavi. In distributed
-    setting, this function only download checkpoint at local rank 0.
+    setting, this function dowload ckpt at all ranks to seperate temp dir.
```
Comment (ZwwWayne) [inline mmcv/runner/checkpoint.py:296]: to different temporary directory

**Review by ZwwWayne — COMMENTED** [review]
LGTM, see if related developers have any comments.
@xvjiarui @innerlee

File: mmcv/runner/checkpoint.py (hunk 1 above)
Comment (innerlee) [inline mmcv/runner/checkpoint.py:296]: typo

**Comment by innerlee** [comment]
This is a waste of bandwidth most of the time haha.

Will these downloaded chkpts be cached anywhere? or they are transient

**Comment by ZwwWayne** [comment]
> This is a waste of bandwidth most of the time haha.
> 
> Will these downloaded chkpts be cached anywhere? or they are transient

A common situation is that we do not know the job is run on how many separated machines without shared storage. In such a case, the simplest way is to download in each rank.

**Review by ZwwWayne — APPROVED** [review]

**Comment by innerlee** [comment]
@ZwwWayne How about making it a two-stage process.

1. Step 1: download ckpt at rank 0
2. Step 2: for all ranks that cannot find the file, download by itself

We can avoid unnecessary downloads altogether, and will have very little influence on user experience.

**Pull request merged**